include/seqan3/alphabet/concept.hpp

```diff
@@ -268,3 +268,71 @@ concept bool alphabet_concept = requires (t t1, t t2)
 };
 
 } // namespace seqan3
+
+// ============================================================================
+// conversion to/from char/rank types
+// ============================================================================
+
+namespace seqan3::detail
+{
+
+//!\addtogroup alphabet
+//!\{
+/*!\brief Specialisation of seqan3::detail::convert with `out_t` == seqan3::underlying_char of the input.
+ * \tparam in_t The type of the input, must satisfy seqan3::alphabet_concept.
+ */
+template <typename in_t>
+    requires alphabet_concept<in_t>
+struct convert<underlying_char_t<in_t>, in_t>
+{
+    //!\brief Implementation of seqan3::detail::convert<underlying_char_t<in_t>, in_t>, calls seqan3::to_char() on the input.
+    static constexpr underlying_char_t<in_t> impl(in_t const & in) noexcept
+    {
+        return to_char(in);
+    }
+};
+
+/*!\brief Specialisation of seqan3::detail::convert with `out_t` == seqan3::underlying_rank of the input.
+ * \tparam in_t The type of the input, must satisfy seqan3::alphabet_concept.
+ */
+template <typename in_t>
+    requires alphabet_concept<in_t>
+struct convert<underlying_rank_t<in_t>, in_t>
+{
+    //!\brief Implementation of seqan3::detail::convert<underlying_rank_t<in_t>, in_t>, calls seqan3::to_rank() on the input.
+    static constexpr underlying_rank_t<in_t> impl(in_t const & in) noexcept
+    {
+        return to_rank(in);
+    }
+};
+
+/*!\brief Specialisation of seqan3::detail::convert with `in_t` == seqan3::underlying_char of the output.
+ * \tparam out_t The type of the output, must satisfy seqan3::alphabet_concept.
+ */
+template <typename out_t>
+    requires alphabet_concept<out_t>
+struct convert<out_t, underlying_char_t<out_t>>
```

Comment:
what if the ```out_t``` is a ```char```?  Then the ```underlying_char_t``` is also char, right?
In that case this overload is ambiguous with [line 286](https://github.com/seqan/seqan3/pull/52/files#diff-c4e11b96ebc75682ef6d95d238c5ab8eR286), isn't it?